Utils/HandlerUtil.py

```diff
@@ -61,26 +61,62 @@
 import base64
 import json
 import time
+
+from xml.etree import ElementTree
 from os.path import join
 from Utils.WAAgentUtil import waagent
 from waagent import LoggerInit
 
 DateTimeFormat = "%Y-%m-%dT%H:%M:%SZ"
 
+MANIFEST_XML = "manifest.xml"
+
 class HandlerContext:
     def __init__(self,name):
         self._name = name
         self._version = '0.0'
         return
 
 class HandlerUtility:
-    def __init__(self, log, error, short_name):
+    def __init__(self, log, error, s_name=None, l_name=None, version=None):
         self._log = log
         self._error = error
-        self._short_name = short_name
 
+        if s_name is None or l_name is None or version is None:
+            (long_name, short_name, ext_version) = self._get_extension_info()
+            self._short_name = short_name
+            self._extension_version = ext_version
+            self._log_prefix = '[%s-%s] ' % (long_name, ext_version)
+        else:
+            self._short_name = s_name
+            self._extension_version = version
+            self._log_prefix = '[%s-%s] ' % (l_name, version)
+
+    def get_extension_version(self):
+        return self._extension_version
+    
     def _get_log_prefix(self):
-        return '[%s-%s]' %(self._context._name, self._context._version)
+        return self._log_prefix if self._log_prefix else '[%s-%s] ' % (self._context._name, self._context._version)
```

Comment:
I don't quite understand the original comment. Again the motivation for this is to retain the existing behavior as is, when the desired extension name/version are not specified at the HandlerUtility construction time, as I wouldn't be too comfortable to change the existing behavior for unknown cases.